### PR TITLE
feat(types): add autogenerated endpoint types

### PR DESCRIPTION
**Pull request by DjDeveloperr**

### CODE REVIEW

**Review by MierenManz — COMMENTED** [review]
LGTM, please check comments

File: scripts.json [hunk 1]
```diff
@@ -2,15 +2,15 @@
   "scripts": {
     "check": {
       "desc": "Type-checks the code",
-      "cmd": "deno run mod.ts"
+      "cmd": "deno check -c Deno.jsonc mod.ts"
```
[the remark this answers is not in the document]
Comment (MierenManz) [reply scripts.json]: this comment didn't get deleted somehow?

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,2 @@
 .vscode/
-.idea/
-.DS_Store
\ No newline at end of file
+scripts/api_docs
```
[the remark this answers is not in the document]
Comment (DjDeveloperr) [reply .gitignore:2]: oops, added back.

File: Deno.jsonc [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
```
[the remark this answers is not in the document]
Comment (DjDeveloperr) [reply Deno.jsonc:1]: i think i'll just let it be in this pr for now

[the remark this answers is not in the document]
Comment (MierenManz) [reply Deno.jsonc:1]: yeah. In the future it would be nice tho.

**Review by MierenManz — APPROVED** [review]
LGTM

**Review by Helloyunho — APPROVED** [review]
lgtm

**Pull request merged**